### PR TITLE
frontend - new block "top statements" introduced

### PR DESCRIPTION
**Pull request by kmoppel**
(similar to "top sprocs") but based on pg_stat_statement data

### CODE REVIEW

File: frontend/src/templates/table_top_statements.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<table class="frontTable noBorder">
+    <thead><th>Name</th><th>Calls</th><th>Total Time</th><th>Avg. Time</th></thead>
+    {% for row in data: %}
+    <tr>
+        <td><a href="/perfstatstatements/graph/{{hostuiname}}/{{row.query_id}}">{{ row.query|e }}</a></td>
+        <td align="right">{{ row.calls }}</td>
```
Comment (avaczi) [inline frontend/src/templates/table_top_statements.html]: the ```align``` attribute is long deprecated, I think: http://www.w3.org/TR/html401/present/graphics.html#h-15.1.2

File: frontend/src/topsprocs.py [hunk 2]
```diff
@@ -7,21 +7,22 @@
...
-    h  = int(m/60)
-    m %= 60
+def makeTimeIntervalReadable(total_millis):
+    total_s = int(total_millis) / 1000.0
+    s = int(total_millis / 1000)
+    m = int(s/60)
+    h = int(m/60)
+
+    if total_millis < 1:
+        return str(int(total_millis * 1000)) + "us"
```
Comment (avaczi) [inline frontend/src/topsprocs.py]: μ as a gift ;)

File: frontend/src/topstatements.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import topsprocs
+import datadb
+
+
+TOP_STATEMENTS_SQL = """
+WITH q_data AS (
+    SELECT
+      ssd_timestamp,
+      ssd_query_id,
+      lower(rtrim(ltrim(regexp_replace(regexp_replace(ssd_query, E'[\\t\\n\\r]+' , ' ', 'g'), E' {2,}', ' ', 'g'))))::varchar(55)||'..' as ssd_query,
```
Comment (avaczi) [inline frontend/src/topstatements.py]: Why not replace with an empty string and omit the trims?

File: frontend/src/topstatements.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+UNION ALL
+SELECT
+  *
+FROM
+ q_total_int1
+UNION ALL
+SELECT
+  *
+FROM
+ q_total_int2
```
Comment (avaczi) [inline frontend/src/topstatements.py]: This query would benefit from some explanation, I guess

File: frontend/src/topsprocs.py (hunk 2 above)
Comment (kmoppel) [inline frontend/src/topsprocs.py]: gift accepted ;)

File: frontend/src/topstatements.py (hunk 3 above)
Comment (kmoppel) [inline frontend/src/topstatements.py]: this would remove single spaces also. i shortened that part though now

Comment (avaczi) [inline frontend/src/topstatements.py]: looks better

File: frontend/src/monitorfrontend.py [hunk 5]
```diff
@@ -162,7 +178,9 @@ def renderTop10LastHours(self, order, hours, hostId, limit):
         return table.render(hostid=hostId, hostuiname=hosts.hostIdToUiShortname(hostId),
                             list=topsprocs.getTop10LastXHours(order, hours, hostId, limit))
 
+    def renderTo10StatementsLastHours(self, hostId, data):
```
Comment (a1exsh) [inline frontend/src/monitorfrontend.py]: should that not be `renderTo_p_10Statements`...

File: frontend/src/topstatements.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+          ssd_query,
+          max(ssd_total_time) - min(ssd_total_time) as total_ms,
+          max(ssd_calls) - min(ssd_calls) as calls
+        FROM
+          q_data
+        WHERE
+          ssd_timestamp > now() - %(interval2)s::interval
+        GROUP BY
+          ssd_query_id, ssd_query
+    ) a
```
Comment (a1exsh) [inline frontend/src/topstatements.py]: Somehow I have a feeling that this sub-select could be moved to a CTE also?

Comment (kmoppel) [inline frontend/src/topstatements.py]: true, made it a bit more readable :)

File: frontend/src/monitorfrontend.py (hunk 5 above)
Comment (kmoppel) [inline frontend/src/monitorfrontend.py]: corrected

**Pull request merged**